engine/vector/src/main/java/io/deephaven/vector/CharVector.java

```diff
@@ -228,4 +240,206 @@ protected final Object writeReplace() {
             return getDirect();
         }
     }
+
+    interface Iterator extends CloseablePrimitiveIteratorOfChar {
+        @Override
+        @FinalDefault
+        default Character next() {
+            return TypeUtils.box(nextChar());
+        }
+
+        @Override
+        @FinalDefault
+        default void forEachRemaining(@NotNull final Consumer<? super Character> action) {
+            forEachRemaining((final char element) -> action.accept(TypeUtils.box(element)));
+        }
+
+        /**
+         * A re-usable, immutable CharacterColumnIterator with no elements.
+         */
+        Iterator EMPTY = new Iterator() {
+            @Override
+            public char nextChar() {
+                throw new NoSuchElementException();
+            }
+
+            @Override
+            public boolean hasNext() {
+                return false;
+            }
+        };
+
+        // region streamAsInt
+        /**
+         * Create an unboxed {@link IntStream} over the remaining elements of this ChunkedCharacterColumnIterator by
+         * casting each element to {@code int} with the appropriate adjustment of
+         * {@link io.deephaven.util.QueryConstants#NULL_CHAR NULL_CHAR} to
+         * {@link io.deephaven.util.QueryConstants#NULL_INT NULL_INT}. The result <em>must</em> be
+         * {@link java.util.stream.BaseStream#close() closed} in order to ensure resources are released. A
+         * try-with-resources block is strongly encouraged.
+         *
+         * @return An unboxed {@link IntStream} over the remaining contents of this iterator. Must be
+         *         {@link Stream#close() closed}.
+         */
+        @Override
+        @FinalDefault
+        default IntStream streamAsInt() {
+            return streamAsInt(
+                    (final char value) -> value == QueryConstants.NULL_CHAR ? QueryConstants.NULL_INT : (int) value);
+        }
+        // endregion streamAsInt
+
+        /**
+         * Get a CharacterColumnIterator with no elements. The result does not need to be {@link #close() closed}.
+         *
+         * @return A CharacterColumnIterator with no elements
+         */
+        static Iterator empty() {
+            return EMPTY;
+        }
+
+        /**
+         * Create a CharacterColumnIterator over an array of {@code char}. The result does not need to be
+         * {@link #close() closed}.
+         *
+         * @param values The elements to iterate
+         * @return A CharacterColumnIterator of {@code values}
+         */
+        static Iterator of(@NotNull final char... values) {
```

Comment:
The duplication from the `CloseablePrimitiveIteratorOfXYZ` interfaces is not awesome. This is just for distinct return types, right? Do we actually need that?